### ISSUE TITLE
Codacy/PMD.ExcessiveClassLength: too long class cgeo.geocaching.maps.mapsforge.v6.NewMap

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#10056]
While working on another issue I had to modify class `cgeo.geocaching.maps.mapsforge.v6.NewMap`.

When checking the related PR by Codacy I got warning `PMD.ExcessiveClassLength` (Avoid really long classes.)

While I agree that this is a valid issue, fixing it vastly exceeds my issue's scope (in fact, I have reduced the length of said class by approx 100 lines with it). Thus this should be fixed up with this dedicated issue.

In the meantime, warning is blocked using `@SuppressWarnings("PMD.ExcessiveClassLength")`

Related question: what is the currently set maximum class length in our Codacy settings? I tried to figure this out in Codacy GUI but couldn't find detail rule configs.

**Comment by bekuno** [comment]
Codacy is a online service, which checks the source code based on different checker and their configuration.
Our pattern list is at https://app.codacy.com/gh/cgeo/cgeo/patterns/list
For the java checks are activated Checkstyle, where we have an own configuration file, and PMD with default.
There is the source for the warning PMD.ExcessiveClassLength (and also for PMD.ExcessiveMethodLength).

A detailed description is at https://pmd.github.io/latest/pmd_rules_apex_design.html#excessiveclasslength .
The default values are
~~~
        setProperty(MINIMUM_DESCRIPTOR, 1000d);
        setProperty(CODECLIMATE_REMEDIATION_MULTIPLIER, 150);
~~~
(I dont know where the multiplier is used for)

We do not have coverd PMD in c:geo.
Do we want it or let it as is or deactivate the PMD in codacy?

**Comment by eddiemuc** [comment]
@bekuno 
Don't get me wrong. I do not want to question the active Codacy rule set with this issue. On the contrary, I think it is valuable and NewMap should be fixed. It is just that I was simply not able to do such a large repair in the context of the other stuff I was currently doing (SAF stuff of course :-)), so I needed to use "SuppressWarning" in order to continue.

This issue is meant as a reminder that there is now such a "SuppressWarning" and some repair needs to be done.

Wrt PMD rules: I think they have some very valuable rules not covered by Checkstyle, mainly the "higher-level" things like NPathComplexity which I consider very important. Wnat would be really nice though is if I could run also the PMD rules locally before committing to a PR, so I would save myself the roundtrip time of committing, pushing and waiting for Codacy-results before continuing. Haven't checked though whether Android Studio has a respective plugin and whether the PMD rules we are using are also available as some sort of "config file" for local usage. Is there anything like this available?

**Comment by bekuno** [comment]
> Wrt PMD rules: I think they have some very valuable rules not covered by Checkstyle, mainly the "higher-level" things like NPathComplexity which I consider very important. Wnat would be really nice though is if I could run also the PMD rules locally before committing to a PR, so I would save myself the roundtrip time of committing, pushing and waiting for Codacy-results before continuing. Haven't checked though whether Android Studio has a respective plugin and whether the PMD rules we are using are also available as some sort of "config file" for local usage. Is there anything like this available?

Yes, you can use PMD in AS and we can it integrate as separate test.
For the first part - there is a IDEA plugin (https://plugins.jetbrains.com/plugin/1137-pmdplugin).
For the second a will open a dedicated issue.

**Comment by moving-bits** [comment]
Won't fix. Old map implementations are deprecated as of release 2025.07.17 and superseded by UnifiedMap.
